lib/node_modules/@stdlib/math/base/special/ellipj/lib/assign.js

```diff
@@ -0,0 +1,237 @@
+		sn = ( tmp4[ 0 ] / tmp4[ 2 ] ) / k1inv;
+		cn = tmp4[ 1 ] / tmp4[ 2 ];
+		dn = 1.0 / tmp4[ 2 ];
+		am = NaN;
+	} else if ( m > 1.0 ) {
+		// A&S 16.11.1 for reciprocal parameter, mapping m > 1 to 0 < mu < 1
+		k = sqrt( m );
+		assign( u * k, 1.0 / m, tmp4, 1, 0 );
+		sn = tmp4[ 0 ] / k;
+		cn = tmp4[ 2 ];
+		dn = tmp4[ 1 ];
+		am = NaN;
+	} else if ( m === 0.0 ) {
+		// A&S table 16.6, limiting case m = 0: circular trigonometric functions
+		sincos( u, tmp2, 1, 0 );
+		sn = tmp2[ 0 ];
+		cn = tmp2[ 1 ];
+		dn = 1.0;
+		am = u;
+	} else if ( m === 1.0 ) {
+		// A&S table 16.6: limiting case m = 1: hyperbolic functions
+		sn = tanh( u );
+		cn = 1.0 / cosh( u );
+		dn = cn;
+		am = atan( sinh( u ) );
+	} else if ( m < SQRT_EPS ) {
+		// A&S 16.13.1 for small-u approximations. Additionally, compute K at extra cost in
+		// Order to ensure returned values are correct outside the range [0, K].
+		K4 = 4.0 * ellipk( m );
+		u0 = ( ( u % K4 ) + K4 ) % K4;
+		sincos( u0, tmp2, 1, 0 );
+		s = tmp2[ 0 ];
+		c = tmp2[ 1 ];
+		f = 0.25 * m * ( u0 - ( s * c ) );
+		sn = s - ( f * c );
+		cn = c + ( f * s );
+		dn = 1.0 - ( 0.5 * m * s * s );
+		am = u - ( 0.25 * m * ( u - ( s * c ) ) );
+	} else if ( m > 1.0 - SQRT_EPS ) {
+		// A&S 16.15.1 - 16.15.4 for near-unity approximations. Additionally, compute K at
+		// Extra cost so that we may reflect as needed to ensure the returned values
+		// Are correct.
+
+		// Reduce by the half-period 2K, centered about u = 0
+		K2 = ellipk( m ) * 2.0;
+		uK2cen = ( u / K2 ) + 0.5;
+		uK2 = K2 * ( ( uK2cen % 1.0 ) - 0.5 );
+
+		// Flip sn and cn in this range to get the reflections correct. We must be careful
+
+		// About precisely reusing uK2cen in order to get the edge cases corect.
+		uK4 = uK2cen % 4;
+		FLG = uK4 >= 1.0 && uK4 < 2.0;
+
+		sinhu = sinh( uK2 );
+		sechu = 1.0 / cosh( uK2 );
+		tanhu = tanh( uK2 );
+		gdu = ( floor( uK2cen ) * PI ) + atan( sinhu );
+
+		a = 0.25 * ( 1.0 - m );
+		b = a * ( sinhu - ( uK2 * sechu ) );
+		sn = tanhu + ( b * sechu );
+		cn = sechu - ( b * tanhu );
+		dn = sechu + ( a * ( sinhu + ( uK2 * sechu ) ) * tanhu );
+		am = gdu + b;
+
+		if (FLG) {
+			sn = -sn;
+			cn = -cn;
+		}
+	} else {
+		// A&S 16.4.1. Compute using the arithmetic-geometric mean.
+		a = 1.0;
+		b = sqrt( 1.0 - m );
+		N = -1;
+		do {
+			N += 1;
+			if ( N > 8 ) {
+				throw new Error('Overflow in ellipj');
+			}
+			atmp = ( a + b ) * 0.5;
+			c = ( a - b ) * 0.5;
+			b = sqrt( a * b );
+			a = atmp;
+			ca[ N ] = c / a;
+		} while ( ca[ N ] >= EPS );
+
+		// A&S 16.4.3:
+		phi1 = ( 1 << N ) * ( u * a );
+		while ( N > 1 ) {
+			N -= 1;
+			phi1 = 0.5 * ( phi1 + asin( ca[ N ] * sin( phi1 ) ) );
+		}
+		phi0 = 0.5 * ( phi1 + asin( ca[ 0 ] * sin( phi1 ) ) );
+
+		am = phi0;
+		sincos( am, tmp2, 1, 0 );
+		sn = tmp2[ 0 ];
+		cn = tmp2[ 1 ];
+
+		// When the denominator is small, switch to the definition of dn in terms
+
+		// Of sn. Otherwise compute dn from the last two iterates.
```

Comment:
```suggestion
		// When the denominator is small, switch to the definition of dn in terms of sn. Otherwise compute dn from the last two iterates.
```